langchain/tools/multi_line_human/tool.py

```diff
@@ -0,0 +1,47 @@
+"""Tool for asking human input."""
+
+from typing import Callable
+
+from pydantic import Field
+
+from langchain.tools.base import BaseTool
+
+
+def _print_func(text: str) -> None:
+    print("\n")
+    print(text)
+
+def input_func():
+    print("Insert your text. Press Ctrl-D (or Ctrl-Z on Windows) to end.")
+    contents = []
+    while True:
+        try:
+            line = input()
+        except EOFError:
+            break
+        contents.append(line)
+    return "\n".join(contents)
+
+class MultiLineHumanInputRun(BaseTool):
+    """Tool that adds the capability to ask user for multi line input."""
+
+    name = "MultiLineHuman"
+    description = (
+        "You can ask a human for guidance when you think you "
+        "got stuck or you are not sure what to do next. "
+        "The input should be a question for the human."
+        "This tool version is suitable when you need answers that span over "
+        "several lines."
+    )
+    prompt_func: Callable[[str], None] = Field(default_factory=lambda: _print_func)
```

Comment:
not super familiar with the mutability of Callables, but do we need to be using default_factory's vs just setting default?

Reply:
TBH, not sure either - it may be fine after all with functions - this code is mostly copied from the Human tool, but modified to offer a different default prompt function.

Original code version: https://github.com/hwchase17/langchain/blob/master/langchain/tools/human/tool.py

Looking in the Python official docs, there's no mention about Callable being mutable and requiring a default factory. Maybe we might want to change it here? (docs: https://docs.python.org/3/library/dataclasses.html#mutable-default-values).

Reply:
@paolorechia so this tool would be equivalent to


```python
from langchain.tools import HumanInputRun

def input_func():
    print("Insert your text. Press Ctrl-D (or Ctrl-Z on Windows) to end.")
    contents = []
    while True:
        try:
            line = input()
        except EOFError:
            break
        contents.append(line)
    return "\n".join(contents)
    
tool = HumanInputRun(input_func=input_func)
```

right?